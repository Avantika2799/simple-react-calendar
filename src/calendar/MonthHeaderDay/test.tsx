import { cleanup, render, RenderResult } from '@testing-library/react'
import React from 'react'

import MonthHeaderDay, { Props } from './MonthHeaderDay'

afterEach(cleanup)

describe('MonthHeaderDay', () => {
  let api: RenderResult, props: Props, mockedCustomRender: any

  beforeEach(() => {
    props = {
      blockClassName: 'test-class-name',
      day: 'Monday',
      isWeekend: false,
    }
  })

  describe('when day is `Monday`', () => {
    beforeEach(() => {
      api = render(<MonthHeaderDay {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when day is `Sunday`', () => {
    beforeEach(() => {
      props = {
        ...props,
        day: 'Sunday',
        isWeekend: true,
      }
      api = render(<MonthHeaderDay {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when render prop has been passed', () => {
    beforeEach(() => {
      mockedCustomRender = jest.fn(() => null)

      props = {
        ...props,
        customRender: mockedCustomRender,
      }
      api = render(<MonthHeaderDay {...props} />)
    })

    test('render prop has been called properly', () => {
      expect(props.customRender).toHaveBeenCalledTimes(1)
      expect(props.customRender).toHaveBeenCalledWith(props)
    })
  })
})
