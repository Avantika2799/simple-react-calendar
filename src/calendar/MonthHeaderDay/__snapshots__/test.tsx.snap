// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthHeaderDay when day is \`Monday\` default render 1`] = `
<div>
  <div
    class="test-class-name-days_of_week_day"
  >
    Monday
  </div>
</div>
`;

exports[`MonthHeaderDay when day is \`Sunday\` default render 1`] = `
<div>
  <div
    class="test-class-name-days_of_week_day is-weekend"
  >
    Sunday
  </div>
</div>
`;
